src/shims/unix/freebsd/sync.rs

```diff
@@ -0,0 +1,216 @@
+                        let umtx_time = match this.read_umtx_time(&umtx_time_place)? {
+                            Some(duration) => duration,
+                            None => {
+                                return this.set_last_error_and_return(LibcError("EINVAL"), dest);
+                            }
+                        };
+
+                        let anchor = if umtx_time.flags == absolute_time_flag {
+                            TimeoutAnchor::Absolute
+                        } else {
+                            TimeoutAnchor::Relative
+                        };
+
+                        Some((TimeoutClock::Monotonic, anchor, umtx_time.timeout))
+                    };
+
+                    let dest = dest.clone();
+                    this.futex_wait(
+                        futex_ref,
+                        u32::MAX, // we set the bitset to include all bits
+                        timeout,
+                        callback!(
+                            @capture<'tcx> {
+                                dest: MPlaceTy<'tcx>,
+                            }
+                            |ecx, unblock: UnblockKind| match unblock {
+                                UnblockKind::Ready => {
+                                    ecx.write_int(0, &dest)
+                                }
+                                UnblockKind::TimedOut => {
+                                    ecx.set_last_error_and_return(LibcError("ETIMEDOUT"), &dest)
+                                }
+                            }
+                        ),
+                    );
+                    interp_ok(())
+                } else {
+                    // The manual doesn't document what should happen if `val` is invalid, so we error out.
+                    this.set_last_error_and_return(LibcError("EINVAL"), dest)
+                }
+            }
+            // UMTX_OP_WAKE has a private variant that enables an optimization that stops it from working across processes.
+            // Miri doesn't support that anyway, so we ignore that variant and use the same implementation for all wake ops.
+            op if op == wake || op == wake_private => {
+                let Some(futex_ref) =
+                    this.get_sync_or_init(obj, |_| FreeBSDFutex { futex: Default::default() })
+                else {
+                    // From Linux implemenation:
+                    // No AllocId, or no live allocation at that AllocId.
+                    // Return an error code. (That seems nicer than silently doing something non-intuitive.)
+                    // This means that if an address gets reused by a new allocation,
+                    // we'll use an independent futex queue for this... that seems acceptable.
+                    return this.set_last_error_and_return(LibcError("EFAULT"), dest);
+                };
+                let futex_ref = futex_ref.futex.clone();
+
+                let count = match val{
```

Comment:
```suggestion
                let count = match val {
```
Seems like rustfmt gave up on your code.^^